pkg/config/config.go

```diff
@@ -732,7 +716,19 @@ func (c *config) Load(path string, fsys fs.FS) error {
 		} else if undecoded := metadata.Undecoded(); len(undecoded) > 0 {
 			fmt.Fprintf(os.Stderr, "Unknown config fields: %+v\n", undecoded)
 		}
+		if base.ProjectId == c.baseConfig.ProjectId {
+			fmt.Fprintf(os.Stderr, "WARN: project_id is missing for remote %s this config won't apply to any branch\n", name)
+		} else {
+			duplicatedRemotesProjectsIds[base.ProjectId] = append(duplicatedRemotesProjectsIds[base.ProjectId], name)
+		}
+		// Check for duplicate project IDs across remotes
+		for projectId, remotes := range duplicatedRemotesProjectsIds {
+			if len(remotes) > 1 {
+				fmt.Fprintf(os.Stderr, "WARN: Multiple remotes (%s) have the same project_id: %s. This may lead to unexpected config override.\n", strings.Join(remotes, ", "), projectId)
+			}
+		}
 		if err := base.Validate(fsys); err != nil {
+			fmt.Fprintf(os.Stderr, "Error with remote config %s\n", name)
```

Comment:
**remark**

While testing, when I set up a remote with `project_id = ""` I had the error raising about the "whole config". Not mentioning the actual issue was inside one of the remote definition. This should help provide better guidance in such cases.